aptos-move/framework/aptos-framework/sources/object_code_deployment.move

```diff
@@ -0,0 +1,73 @@
+module aptos_framework::object_code_deployment {
+    use std::bcs;
+    use std::features;
+    use std::signer;
+    use std::string;
+    use std::vector;
+    use aptos_framework::account;
+    use aptos_framework::code;
+    use aptos_framework::code::PackageRegistry;
+    use aptos_framework::object;
+    use aptos_framework::object::{ExtendRef, Object};
+
+    /// Object code deployment not supported.
+    const EOBJECT_CODE_DEPLOYMENT_NOT_SUPPORTED: u64 = 0x1;
+    /// Not the owner of the `PublisherRef`
+    const ENOT_OWNER: u64 = 0x2;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct PublisherRef has key {
+        extend_ref: ExtendRef,
+    }
+
+    /// Create a new object to host the code, and `PublisherRef` if the code is upgradeable,
+    /// Send `PublisherRef` to object signer.
+    public entry fun publish(
+        publisher: &signer,
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
```

Comment:
Why is this a vector of vectors? Is it one vector per module?

If I'm not wrong, we expect people to look at this code, it's not an "internal" Move module per se. So a comment explaining this would be great, even if the inner functions explain what's going on.

Reply:
That's right, vector per module. Will add more details on the documentation to clarify this, thanks for pointing it out!